src/matlab2tikz.m

```diff
@@ -2503,6 +2503,28 @@ case size(cData,2) % specific y-data is given
     end
 end
 % ==============================================================================
+% Function `handle` is not yet implemented in Octave.
+% Consequently the plot type needs to be guessed. See #645.
+% If the type can not be determined reliably, 'unknown' will be set.
+function cl = guessOctavePlotType(h)
+    properties = get(h);
+    % scatter plots
+    if isfield(properties,'marker') && ...
```

Comment:
Just a few minor things I think of
 - Doesn't Octave have the `type` property as MATLAB does in the HG1 framework?
 - I think we can make this code clearer by introducing a function `hasProperties(h, propertynames)` that would check the presence of those properties and preferably do so case-insensitively. Then the fragment would become something like:

```matlab
if hasProperties(h, {'Marker', 'SizeData', 'CData'})
    cl = 'specgrap.scattergroup';
elseif ...
    ...
end
```